libpod/reset.go

```diff
@@ -77,18 +77,35 @@ func (r *Runtime) Reset(ctx context.Context) error {
 		}
 	}
 
+	xdgRuntimeDir := os.Getenv("XDG_RUNTIME_DIR")
 	_, prevError := r.store.Shutdown(true)
-	if err := os.RemoveAll(r.store.GraphRoot()); err != nil {
+	graphRoot := r.store.GraphRoot()
+	if graphRoot == xdgRuntimeDir {
 		if prevError != nil {
 			logrus.Error(prevError)
 		}
-		prevError = err
+		prevError = errors.Errorf("failed to remove runtime graph root dir %s, since it is the same XDG_RUNTIME_DIR", graphRoot)
+	} else {
+		if err := os.RemoveAll(graphRoot); err != nil {
+			if prevError != nil {
+				logrus.Error(prevError)
+			}
+			prevError = err
+		}
 	}
-	if err := os.RemoveAll(r.store.RunRoot()); err != nil {
+	runRoot := r.store.RunRoot()
+	if runRoot == xdgRuntimeDir {
```

Comment:
It might be worth checking whether the "run root" is correctly prefixed, if it is a child of xdgRuntimeDir.

Reply:
Why would that matter?

Reply:
It matters if you can only do string comparisons. The new default configuration is `$XDG_RUNTIME_DIR/overlays`, it used to be `$XDG_RUNTIME_DIR`. You still don't want to nuke `$XDG_RUNTIME_DIR/`.

Reply:
Right the case above will only remove $XDG_RUNTIME_DIR/overlays which is what we want.  It really does not mapper if XDG_RUNTIME_DIR/overlays or /random/path/overlays